src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadCaseFilesAboutToSubmitHandler.java

```diff
@@ -206,117 +155,58 @@ private boolean isTypeValidForOtherDocs(String caseDocumentType) {
-        }
-    }
 
     private void filterConfidentialDocs(List<ContestedUploadedDocumentData> uploadedDocuments,
                                         Map<String, Object> caseData,
-                                        String collection,
-                                        String party) {
+                                        String collectionName) {
 
         log.info("UploadDocuments Collection: {}", uploadedDocuments);
         List<ContestedUploadedDocumentData> confidentialFiltered = uploadedDocuments.stream()
             .filter(d -> d.getUploadedCaseDocument().getCaseDocuments() != null
-                && d.getUploadedCaseDocument().getCaseDocumentParty() != null
-                && d.getUploadedCaseDocument().getCaseDocumentParty().equals(party))
-            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentType() != null
+                && d.getUploadedCaseDocument().getCaseDocumentType() != null
                 && d.getUploadedCaseDocument().getCaseDocumentConfidential() != null
                 && d.getUploadedCaseDocument().getCaseDocumentConfidential().equalsIgnoreCase("Yes"))
             .collect(Collectors.toList());
 
-        List<ContestedUploadedDocumentData> confidentialDocsCollection = getDocumentCollection(caseData, collection);
+        List<ContestedUploadedDocumentData> confidentialDocsCollection = getDocumentCollection(caseData, collectionName);
         confidentialDocsCollection.addAll(confidentialFiltered);
         log.info("Adding items: {}, to Confidential Documents Collection", confidentialFiltered);
         uploadedDocuments.removeAll(confidentialFiltered);
 
         if (!confidentialDocsCollection.isEmpty()) {
-            caseData.put(collection, confidentialDocsCollection);
+            List<ConfidentialUploadedDocumentData> confidentialDocs = confidentialDocsCollection.stream().map(
+                doc -> ConfidentialUploadedDocumentData.builder()
+                    .confidentialUploadedDocument(ConfidentialUploadedDocument.builder()
+                        .documentFileName(doc.getUploadedCaseDocument().getCaseDocuments().getDocumentFilename())
+                        .documentComment(doc.getUploadedCaseDocument().getHearingDetails())
+                        .documentLink(doc.getUploadedCaseDocument().getCaseDocuments())
+                        .build()).build()).collect((Collectors.toList()));
+            caseData.put(collectionName, confidentialDocs);
+        }
+    }
+
+    private void filterFdrDocs(List<ContestedUploadedDocumentData> uploadedDocuments,
+                               Map<String, Object> caseData,
+                               String collectionName) {
+
+        log.info("UploadDocuments Collection: {}", uploadedDocuments);
+        List<ContestedUploadedDocumentData> fdrFiltered = uploadedDocuments.stream()
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocuments() != null
```

Comment:
probably best to extract this predicate to a method, especially with the duplication on getUploadedCaseDocument?